src/XamlX/Transform/XamlContextBase.cs

```diff
@@ -12,6 +12,10 @@ class XamlContextBase
     {
         private readonly Dictionary<Type, object> _items = new Dictionary<Type, object>();
         private readonly List<IXamlAstNode> _parentNodes = new List<IXamlAstNode>();
+        private int _idCount;
+        
+        public int GetNextUniqueContextId () => ++_idCount;
```

Comment:
If you need thread safety here consider using https://docs.microsoft.com/en-us/dotnet/api/system.threading.interlocked.increment?view=netcore-3.1

Reply:
We don't, the underlying MSIL manipulation API isn't thread-safe anyway.